Closes #3060 Remove `ex_single` dataset to reduce package size [create website] (#3061)
pharmaverse/admiral: vignettes/occds.Rmd

ae <- pharmaversesdtm::ae
adsl <- admiral::admiral_adsl
ex <- pharmaversesdtm::ex

> 

Reviewer: ```suggestion
```

ae <- pharmaversesdtm::ae
adsl <- admiral::admiral_adsl
ex <- pharmaversesdtm::ex

ae <- convert_blanks_to_na(ae)

> # Create single dose dataset: expand multi-day dose intervals into one row per day

Reviewer 2: I would put the processing of the `ex` dataset in a separate chunk and add some explanations why this is done and that it depends on the data collection whether the `create_single_dose_dataset()` call is necessary. Maybe also a note that for ongoing studies it may be necessary to impute missing dosing end dates, e.g., with the data cut off date.

We could also move it to the "Derive Date/Date-time of Last Dose" section. Then it may be clearer and easier to explain.

What do you think?

ex_single <- convert_blanks_to_na(ex) %>%
  # Filter to doses relevant for this study (placebo and active doses)

>   filter(EXDOSE %in% c(0, 54)) %>%

Reviewer 2: Do we need this?

If it doesn't cause issues, I would remove it to keep the code simple. Excluding specific doses looks strange to me anyway.

  derive_vars_dt(dtc = EXSTDTC, new_vars_prefix = "EXST") %>%
  derive_vars_dt(dtc = EXENDTC, new_vars_prefix = "EXEN") %>%
  filter(!is.na(EXSTDT), !is.na(EXENDT)) %>%
  create_single_dose_dataset(
    dose_freq = EXDOSFRQ,
    start_date = EXSTDT,
    end_date = EXENDT,
    keep_source_vars = exprs(
      STUDYID, USUBJID, EXTRT, EXDOSE, EXDOSU, EXDOSFRQ, EXSTDT, EXENDT
    )
  ) %>%
  # Convert date objects back to character DTC format for use in derive_vars_dtm()
  mutate(
    EXSTDTC = as.character(EXSTDT),
    EXENDTC = as.character(EXENDT)

>   )

Reviewer 2: This should be simplified. We create `*DTC` variables here and use them only to convert them to `*DT` variables although they are already in `ex_single`. I.e., we could remove this conversion step and the conversion steps where `ex_single` is used.

## Derive Date/Date-time of Last Dose {#last_dose}

> Before deriving the last dose date, an `ex_single` dataset is created from the

Reviewer 2: I would suggest:
```suggestion
Before deriving the last dose date, it may be necessary to create an `ex_single` dataset from the
```
because the single dose dataset is only required for some studies.

`EX` domain. If the exposure dataset contains multi-day dosing records (e.g.,
one record per treatment period rather than one record per dose), use

> `create_single_dose_dataset()` to expand them into one record per day. Whether

Reviewer 2: ```suggestion
`create_single_dose_dataset()` to expand them into one record per dose. Whether
```

this step is necessary depends on how dosing data were collected in your study.
For ongoing studies, you may also need to impute missing end dates (e.g., with
the data cut-off date) before calling `create_single_dose_dataset()`. For a detailed discussion of dose expansion, including guidance on handling

> missing end dates for ongoing studies, see `vignette("pk_adnca", package = "admiral")`.

Reviewer 2: I couldn't find anything regarding cut-off date in the PK vignette. However, we have an example in the `create_single_dose_dataset()` documentation. Thus I would rather refer to these examples and use our custom roclet tags for the `create_single_dose_dataset()` examples.

> 

Reviewer 2: I would add something like "The test data contains one record per treatment period and the dose frequency is daily. Thus the following `create_single_dose_dataset()` call creates one record per day."

```{r eval=TRUE}
ex_single <- convert_blanks_to_na(ex) %>%
  derive_vars_dt(dtc = EXSTDTC, new_vars_prefix = "EXST") %>%

>   derive_vars_dt(dtc = EXENDTC, new_vars_prefix = "EXEN") %>%

Reviewer 2: I would use `derive_vars_dtm()` because we use only the datetime variables in the derivations and if necessary, imputations could be specified here.

  filter(!is.na(EXSTDT), !is.na(EXENDT)) %>%
  create_single_dose_dataset(
    dose_freq = EXDOSFRQ,
    start_date = EXSTDT,
    end_date = EXENDT,
    keep_source_vars = exprs(
      STUDYID, USUBJID, EXTRT, EXDOSE, EXDOSU, EXDOSFRQ, EXSTDT, EXENDT
    )
  ) %>%
  mutate(
    EXSTDTM = as_datetime(EXSTDT),
    EXENDTM = as_datetime(EXENDT)

>   )

Reviewer 2: If `derive_vars_dtm()` is used above, this call can be removed. I think it would also be clearer as imputations could be specified directly in `derive_vars_dtm()`.

Furthermore, using `derive_vars_dt()` and `as_datetime()` has the side effect that it removes the time information from the dosing records. I'm not sure if this was intended.

## Derive Date/Date-time of Last Dose {#last_dose}

Before deriving the last dose date, it may be necessary to create an `ex_single` dataset from the
`EX` domain. If the exposure dataset contains multi-day dosing records (e.g.,
one record per treatment period rather than one record per dose), use
`create_single_dose_dataset()` to expand them into one record per dose. Whether
this step is necessary depends on how dosing data were collected in your study.
For ongoing studies, you may also need to impute missing end dates (e.g., with

> the data cut-off date) before calling `create_single_dose_dataset()`. For

Reviewer: Do we cover this somewhere else in our docs?

examples including handling of missing end dates, see `?create_single_dose_dataset`.

The test data contains one record per treatment period and the dose frequency is
daily (`QD`). The following call creates one record per day.

```{r eval=TRUE}
ex_single <- convert_blanks_to_na(ex) %>%

>   derive_vars_dtm(dtc = EXSTDTC, new_vars_prefix = "EXST", flag_imputation = "none") %>%

Reviewer: ```suggestion
ex_single <- ex %>%
derive_vars_dtm(dtc = EXSTDTC, new_vars_prefix = "EXST", flag_imputation = "none") %>%
```

can remove as we already did this step

  derive_vars_dtm(
    dtc = EXENDTC,
    new_vars_prefix = "EXEN",
    time_imputation = "last",
    flag_imputation = "none"
  ) %>%
  derive_vars_dtm_to_dt(exprs(EXSTDTM, EXENDTM)) %>%
  filter(!is.na(EXSTDT), !is.na(EXENDT)) %>%
  create_single_dose_dataset(
    dose_freq = EXDOSFRQ,
    start_date = EXSTDT,
    end_date = EXENDT,
    keep_source_vars = exprs(
      STUDYID, USUBJID, EXTRT, EXDOSE, EXDOSU, EXDOSFRQ, EXSTDT, EXENDT
    )
  ) %>%
  mutate(
    EXSTDTM = as_datetime(EXSTDT),
    EXENDTM = as_datetime(EXENDT)

>   )

Reviewer 2: I would suggest:
```suggestion
ex_single <- ex %>%
  derive_vars_dtm(
    dtc = EXSTDTC,
    new_vars_prefix = "EXST",
    flag_imputation = "none"
  ) %>%
  derive_vars_dtm(
    dtc = EXENDTC,
    new_vars_prefix = "EXEN",
    time_imputation = "last",
    flag_imputation = "none"
  ) %>%
  filter(!is.na(EXSTDTM), !is.na(EXENDTM)) %>%
  create_single_dose_dataset(
    dose_freq = EXDOSFRQ,
    start_date = EXSTDTM,
    end_date = EXENDTM,
    keep_source_vars = exprs(
      STUDYID, USUBJID, EXTRT, EXDOSE, EXDOSU, EXDOSFRQ, EXSTDTM, EXENDTM
    )
  )
```

Reviewer 3: @bundfussr I think this is the correct way, otherwise the original versions of `EXSTDTM` and ` EXENDTM` are kept
```
  create_single_dose_dataset(
    dose_freq = EXDOSFRQ,
    start_date = EXSTDT,
    start_datetime = EXSTDTM,
    end_date = EXENDT,
    end_datetime = EXENDTM,
    keep_source_vars = exprs(
      STUDYID, USUBJID, EXTRT, EXDOSE, EXDOSU, EXDOSFRQ, EXSTDT, EXENDT, EXSTDTM,
      EXENDTM,
    )
  )
```

Reviewer 2: Ah yes, you are right.
